tests/unittests/BackendTest.cpp

```diff
@@ -254,6 +254,55 @@ TEST_P(BackendTest, simplePlaceholderValue) {
   EXPECT_TRUE(STensor->isEqual(data));
 }
 
+/// Add and compile a network, then add and compile another so that the first
+/// CompiledFunction does not know about every Placeholder in the module.
+TEST_P(BackendTest, compileThenAddNetwork) {
+  Tensor inputs(ElemKind::FloatTy, {1, 10, 10, 3});
+  PlaceholderBindings bindings1, bindings2;
+
+  auto &mod = EE_.getModule();
+
+  // Create a simple graph that uses some placeholders.
+  Function *F = mod.createFunction("main");
+  auto *input =
+      mod.createPlaceholder(ElemKind::FloatTy, {1, 10, 10, 3}, "in", false);
+
+  auto *ex = mod.createPlaceholder(ElemKind::Int64ITy, {1, 1}, "exp", false);
+
+  auto *FC = F->createFullyConnected(bindings1, "FC", input, 30);
+  auto *RL = F->createRELU("RL2", FC);
+  auto *SM = F->createSoftMax("sm", RL, ex);
+  auto *S = F->createSave("ret", SM);
+
+  EE_.compile(CompilationMode::Infer, F);
+
+  // Recreate that graph in a different Function.
```

Comment:
Why not use `Function::clone` here instead of explicitly building the exact same graph again? It also takes an optional `map<Node*, Node*>` that you can use to get references to the output of one `Placeholder` replaced with references to the output of another (so that you can make sure the cloned Function connects to different `Placeholders`).

You can get a pointer to the final `Save` node using `Function::getNodeByName` or `Function::getNodes`.

Reply:
But given that you are using the version of `Function::createFullyConnected` that accepts a reference to a `PlaceholderBindings` object as an argument and adds bindings for the weight/bias `Placeholders` it creates, doing what I suggested might be more trouble than it's worth unless you choose simpler models.

Reply:
I think i'd prefer not to clone for this example for that reason. The goal here is to test that two totally different functions work in the same module.